src/lib/storage/table.cpp

```diff
@@ -132,6 +133,17 @@ std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
   return _chunks[chunk_id];
 }
 
+void Table::delete_chunk(ChunkID chunk_id) {
+  DebugAssert(chunk_id < _chunks.size(), "ChunkID " + std::to_string(chunk_id) + " out of range");
+  DebugAssert(_chunks[chunk_id]->invalid_row_count() == _chunks[chunk_id]->size(),
+              "Physical delete of chunk prevented: Chunk needs to be fully invalidated before.");
```

Comment:
Is `invalid_row_count` thread-safe? If not, this might never be true. Also, we might have other implications that we should be aware of.

Reply:
Currently, `invalid_row_count` is not thread-safe. So this `DebugAssert` is a problem. Right!
We have two possibilities:
1. Implement locks to make it thread-safe. This might cost us some performance.
2. Accept that `invalid_row_count` is not thread-safe and live with it as a lazy approach. Thus, we'd have to remove this `DebugAssert`.

What do you suggest?